libcxx/test/std/time/time.zone/time.zone.leap/assign.copy.pass.cpp

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+// UNSUPPORTED: no-filesystem, no-localization, no-tzdb
+
+// XFAIL: libcpp-has-no-incomplete-tzdb
+// XFAIL: availability-tzdb-missing
+
+// <chrono>
+
+// class leap_second
+// {
+//   leap_second& operator=(const leap_second&) = default;
+//
+//   ...
+// };
+
+#include <chrono>
+#include <memory>
+#include <type_traits>
+#include <cassert>
+
+// Add the include path required by test_chrono_leap_second.h when using libc++.
+// ADDITIONAL_COMPILE_FLAGS(stdlib=libc++): -I %S/../../../../../src/include
```

Comment:
I think we need tests that check the behavior of assignment from rvalue (the behavior is that it should copy rather than move).

Same thing for the copy constructor.

Reply:
Note I test the constructor only for libc++. Per https://eel.is/c++draft/time.zone.leap.overview
```
  class leap_second {
  public:
    leap_second(const leap_second&)            = default;
    leap_second& operator=(const leap_second&) = default;

    // unspecified additional constructors

    constexpr sys_seconds date() const noexcept;
    constexpr seconds value() const noexcept;
  };
```
implementations are allowed to add a move constructor.

Reply:
I disagree that implementations are allowed to add move constructors. The leap second object does not have a "moved from" state, and implementations should not have different behavior here.

The comment is likely referring to the private or internal constructors used to initially construct the object. 

So I still think we need the additional tests.

Reply:
Oh, can you double check we have tests that it isn't default constructable?

Reply:
> I disagree that implementations are allowed to add move constructors. The leap second object does not have a "moved from" state, and implementations should not have different behavior here.

It would be Standard conforming, from a QoI PoV I expect nobody to do it.
I can add the test with a libc++ specific `static_assert`, the same for default constructible.